### ISSUE TITLE
AgdaStdlib – permission denied in build

### ISSUE THREAD

**Issue by laMudri** [NixOS/nixpkgs#15306]
## Issue description
As of Agda 2.5.1 (which comes on the latest nixos-unstable release), both nixpkgs' agda-stdlib (from git nominally at version 2.4.2.3) and an updated version (v0.12, the most recent .tar.gz release) fail to build. Some modules are checked successfully, but it gets stuck at `Agda.Builtin.Unit` (the first builtin module it finds). For this module only, it tries to write to `/nix/store/`.

Output:
```
building path(s) ‘/nix/store/p8nqivxn3zr6mmwis8xgyshjgqhdxgf8-agda-stdlib-v0.12’
unpacking sources
unpacking source archive /nix/store/yr0h4ia34i9q6mqnslqgm1a5draksxyg-v0.12.tar.gz
source root is agda-stdlib-0.12
setting SOURCE_DATE_EPOCH to timestamp 1460834616 of file agda-stdlib-0.12/.travis.yml
patching sources
configuring
no configure script, doing nothing
building
Checking Everything (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/Everything.agda).
 Checking Algebra (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Algebra.agda).
  Checking Relation.Binary (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Relation/Binary.agda).
   Checking Data.Product (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Product.agda).
    Checking Function (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Function.agda).
     Checking Level (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Level.agda).
     Finished Level.
    Finished Function.
    Checking Relation.Nullary (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Relation/Nullary.agda).
     Checking Data.Empty (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Empty.agda).
     Finished Data.Empty.
    Finished Relation.Nullary.
   Finished Data.Product.
   Checking Data.Sum (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Sum.agda).
    Checking Data.Maybe.Base (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Maybe/Base.agda).
     Checking Data.Bool.Base (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Bool/Base.agda).
      Checking Data.Unit.Base (/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Unit/Base.agda).
       Checking Agda.Builtin.Unit (/nix/store/qpzrr8ymvh7ay5z96fci0qp71n4x990k-Agda-2.5.1/share/x86_64-linux-ghc-7.10.3/Agda-2.5.1/lib/prim/Agda/Builtin/Unit.agda).
Failed to write interface /nix/store/qpzrr8ymvh7ay5z96fci0qp71n4x990k-Agda-2.5.1/share/x86_64-linux-ghc-7.10.3/Agda-2.5.1/lib/prim/Agda/Builtin/Unit.agdai.
/tmp/nix-build-agda-stdlib-v0.12.drv-0/agda-stdlib-0.12/src/Data/Unit/Base.agda:13,13-30
/nix/store/qpzrr8ymvh7ay5z96fci0qp71n4x990k-Agda-2.5.1/share/x86_64-linux-ghc-7.10.3/Agda-2.5.1/lib/prim/Agda/Builtin/Unit.agdai:
openBinaryFile: permission denied (Permission denied)
builder for ‘/nix/store/z7nzphrz1m57mzp7516w53sl5nfa7a81-agda-stdlib-v0.12.drv’ failed with exit code 1
error: build of ‘/nix/store/z7nzphrz1m57mzp7516w53sl5nfa7a81-agda-stdlib-v0.12.drv’ failed
```

### Steps to reproduce
```
nix-build -A AgdaStdlib
```

## Technical details

* System: 16.09.git.39b3345 (Flounder)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09.git.39b3345"

**Comment by laMudri** [comment]
`agdaPrelude` has the same issue, meaning that there has probably been some change in Agda. The problem seems to be that `agda` writes a `.agdai` file for each `.agda` file it checks, and some files it's checking are in the Nix store (specifically, in `/nix/store/...-Agda-2.5.1`). I think it's to do with [this](https://github.com/agda/agda/blob/master/CHANGELOG#L239). Maybe it would work if these were checked when building `Agda`.

**Comment by laMudri** [comment]
Current status: [internal error](https://github.com/agda/agda/issues/1988).

**Comment by laMudri** [comment]
Fixed by #15595.

**Comment by joachifm** [comment]
Let's keep this open until the fix lands in master.
